run_unittests.py

```diff
@@ -523,16 +523,18 @@ def _run(self, command, workdir=None):
         Run a command while printing the stdout and stderr to stdout,
         and also return a copy of it
         '''
-        p = subprocess.Popen(command, stdout=subprocess.PIPE,
-                             stderr=subprocess.STDOUT, env=os.environ.copy(),
-                             universal_newlines=True, cwd=workdir)
-        output = p.communicate()[0]
-        print(output)
+        # If this call hangs CI will just abort. It is very hard to distinguish
+        # between CI issue and test bug in that case. Set timeout and fail loud
+        # instead.
+        p = subprocess.run(command, stdout=subprocess.PIPE,
+                           stderr=subprocess.STDOUT, env=os.environ.copy(),
+                           universal_newlines=True, cwd=workdir, timeout=60 * 5)
```

Comment:
I assume we could actually use a much lower timeout, any test taking longer than 30 seconds is probably defective

Reply:
That is true, on the other hand if for some reason CI container is getting slow we might end up being more flaky than necessary. We should pick value less than CI timeout for no output, but I don't see a particular reason to be extremely strict.

Reply:
as you think is best, I tend to prefer reasonably fast failures :)

Reply:
as you think is best, I tend to prefer reasonably fast failures :)